src/store/modules/position.store.js

```diff
@@ -264,6 +259,37 @@ const actions = {
             )
         }
         if (matchingProjection) {
+            const oldProjection = state.projection
+            // reprojecting the center of the map
+            const [x, y] = proj4(oldProjection.epsg, matchingProjection.epsg, state.center)
+            commit('setCenter', { x, y })
+            // adapting the zoom level (if needed)
+            if (
+                oldProjection instanceof StandardCoordinateSystem &&
+                matchingProjection instanceof CustomCoordinateSystem
+            ) {
+                commit('setZoom', matchingProjection.transformStandardZoomLevelToCustom(state.zoom))
+            }
+            if (
+                oldProjection instanceof CustomCoordinateSystem &&
+                matchingProjection instanceof StandardCoordinateSystem
+            ) {
+                commit('setZoom', oldProjection.transformCustomZoomLevelToStandard(state.zoom))
+            }
+            if (
+                oldProjection instanceof CustomCoordinateSystem &&
+                matchingProjection instanceof CustomCoordinateSystem &&
+                oldProjection.epsg !== matchingProjection.epsg
```

Comment:
See comment above, I guess if the projection did not change we don't need to change the center nor the zoom ?